### PR TITLE
Support for bazel buildsystem

### PR DESCRIPTION
**Pull request by mjcarroll**

### CODE REVIEW

**Comment by j-rivero** [comment]
Do we have any option of running a testing build in github action or other CI?

File: OgreMain/src/OgreRoot.cpp [hunk 1]
```diff
@@ -1465,9 +1467,11 @@ namespace Ogre {
     //-----------------------------------------------------------------------
     void Root::loadPlugin(const String& pluginName)
     {
+      std::cout << "Load: " << pluginName << std::endl;
```
Comment (j-rivero) [inline OgreMain/src/OgreRoot.cpp:1470]: Do you want to keep this code change or is it a leftover from a debugging session?

File: BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    ],
+    deps = [
+        ":ogre_main_headers",
+        "@X//:SM",
+        "@X//:Xaw",
+        "@X//:Xrandr",
+        "@freeimage",
+        "@gl//:GL",
+        "@gl//:GLU",
+        "@gl//:glut",
```
Comment (mahiuchun) [inline BUILD.bazel:206]: do we actually use `glut`?

File: OgreMain/src/OgreRoot.cpp (hunk 1 above)
Comment (mjcarroll) [reply OgreMain/src/OgreRoot.cpp]: Removed.

File: BUILD.bazel (hunk 2 above)
Comment (mjcarroll) [reply BUILD.bazel]: We had it in the CMake, but it appears to work fine without it, so I have removed it.

**Pull request merged**